tests/NCronJob.Tests/NCronJobIntegrationTests.cs

```diff
@@ -163,6 +165,36 @@ public async Task NotRegisteredJobShouldNotAbortOtherRuns()
         jobFinished.ShouldBeTrue();
     }
 
+    [Fact]
+    public void ThrowIfJobWithDependenciesIsNotRegistered()
+    {
+        ServiceCollection
+            .AddNCronJob(n => n.AddJob<JobWithDependency>(p => p.WithCronExpression("* * * * *")));
+        var provider = CreateServiceProvider();
+
+        Assert.Throws<InvalidOperationException>(() =>
+        {
+            using var executor = new JobExecutor(provider, NullLogger<JobExecutor>.Instance);
+            executor.RunJob(new RegistryEntry(typeof(JobWithDependency), new JobExecutionContext(null), null), CancellationToken.None);
+
+        });
+    }
+
+    [Fact]
+    public async Task NotThrowIfJobWithDependenciesRegistered()
```

Comment:
That is basically tested by many other tests indirectly. I would remove that test.

Reply:
Good point - it's removed